src/main.rs

```diff
@@ -49,7 +49,19 @@ fn calculate_safety_status(
     has_carrot: bool,
     friends_nearby: i32,
 ) -> bool {
-    todo!()
+    if !wolves_nearby {
+        if day_time {
+            return true;
+        }
+        else if has_carrot {
+            return true;
+        }
+        else if friends_nearby > 3 {
+            return true;
+        }
+    }
+    return false;
```

Comment:
This is pretty good! But you can actually simplify this quite a lot by using boolean logic all on one line. It might look something like `a || (b && c)`.